### ISSUE TITLE
Getting the MAC table fails on recent ONIE switches

### ISSUE THREAD

**Issue by kcgthb** [xcat2/xcat-core#7463]
As describe in #7462 , getting the MAC table from ONIE switches is initially attempted via paswordless SSH:

The issue is that the command used in [MacMap.pm](https://github.com/xcat2/xcat-core/blob/master/perl-xCAT/xCAT/MacMap.pm) doesn't seem to work on more recent Cumulus versions (I'm using 5.9), likely due to a path change:
https://github.com/xcat2/xcat-core/blob/c8cb1d6a38f9dc204313ff7b12b055f03ca233e6/perl-xCAT/xCAT/MacMap.pm#L726

```
$ ssh $switch 'bridge fdb show'
Debian GNU/Linux 12
bash: line 1: bridge: command not found
```

Although it would be possible to force the full path:
```
$ ssh $switch '/usr/sbin/bridge fdb show'
Debian GNU/Linux 12
b0:cf:0e:0d:11:72 dev swp7 master br_default permanent
b0:cf:0e:0d:11:73 dev swp8 master br_default permanent
b0:cf:0e:0d:11:28 dev swp9 master br_default permanent
[...]
```
that's probably not a good fix, as it may not survive new path changes in the future.

But the real issue here is that the "command not found" error is masked by the `grep` and `tr` pipes, and the initial error code is not returned through SSH. So, from the script's perspective, the `ssh` command succeeds, but doesn't return any MAC address, so `refresh_switch()` returns an empty list. And nodes cannot be discovered.

The `bridge` command is not found because running `ssh host cmd` will not start a login shell, so the remote `PATH` variable is not properly set. So maybe the remote command should be wrapped in a login shell, to make sure all the environment is set correctly and commands are found?

I'll propose a PR shortly.

**Comment by kcgthb** [comment]
Proposed fix in #7464

**Comment by Obihoernchen** [comment]
Merged
